CheckSammy.py

```diff
@@ -438,86 +444,85 @@ class CheckSammy():
     def join_path(self,a,b):
         return a + '/' + b
 
-    def calc_md5(self, file):
+    def calculate_md5(self, file):
         # Calculates the md5 for the selected file.
         h = hashlib.md5()
 
-        with open(file, 'rb') as tohash:
-            for chunk in iter(lambda: tohash.read(4096), b''):
+        with open(file, 'rb') as file_data:
+            for chunk in iter(lambda: file_data.read(4096), b''):
                 h.update(chunk)
 
         return(h.hexdigest())
 
-    def check_md5(self, ff, switch=0):
-        # Calculates a new checksum and confronts it with the one stored in the json.
+    def check_md5(self, path, switch=0):
+        # Calculates a new checksum and compares it with the one stored in the json.
+        target_name = os.path.basename(path) # File or directory
+
         if switch == 0:
             try:
-                old_cs = json.load(open(ff + '.md5'))[os.path.basename(ff)]
-                new_cs = self.calc_md5(ff)
-                if old_cs == new_cs:
-                    puppy.checked['Ok'].append(os.path.basename(ff))
+                previous_checksum = json.load(open(path + '.md5'))[target_name]
+                new_checksum = self.calculate_md5(path)
+                if previous_checksum == new_checksum:
+                    puppy.checked['Ok'].append(target_name)
                 else:
-                    puppy.checked['Corrupted'].append(os.path.basename(ff))
+                    puppy.checked['Corrupted'].append(target_name)
             except FileNotFoundError:
-                puppy.checked['No md5'].append(os.path.basename(ff) + '.md5')
+                puppy.checked['No md5'].append(target_name + '.md5')
 
         elif switch == 1:
-            new_hash_dict = {}
-            new_hash_dict['PARENT FOLDER'] = os.path.basename(ff)
+            new_dict = {'PARENT FOLDER': target_name}
             try:
-                old_hash_dict = json.load(open(ff + '.md5'))
+                previous_dict = json.load(open(path + '.md5'))
```

Comment:
Hashes and dicts are essentially synonyms (in Python they're called dicts, in Ruby hashes, etc)  ;-)